.github/workflows/docs-gen-and-push.yml

```diff
@@ -8,6 +8,7 @@ on:
   push:
     branches:
       - "release-*"
+      - "doc-*"
```

Comment:
Why do we need to restrict to specific branch name?  We already have paths rule, If someone is changing files under docs/ I expect that user to want to see the generated site even if it's not a "doc-"  release

Reply:
The idea here is that this rendering does not happen normally, it only happens when the contributor requests it. This saves the costs of doing renderings that have little value. We have been doing many PRs that involve doc updates without people asking for an automatic rendering.

The contributor requests rendering by naming their branch something that starts with "doc-".

`"doc"- release` is not a thing. I am not sure I understand.

Elsewhere you expressed concern over the costs of the rendering. The control exerted here limits that.